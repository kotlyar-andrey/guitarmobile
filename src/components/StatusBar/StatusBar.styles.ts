import {StatusBarStyle} from 'react-native';
import {Theme} from '~/theming/interfaces';

interface Props {
  backgroundColor: string;
  barStyle?: StatusBarStyle;
}

const createStyles = (theme: Theme) => {
  const styles: Props = {
    backgroundColor: theme.colors.background,
    barStyle: theme.dark ? 'light-content' : 'dark-content',
  };
  return styles;
};
export default createStyles;
