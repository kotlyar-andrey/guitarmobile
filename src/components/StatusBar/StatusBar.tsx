import {StatusBar as RNStatusBar} from 'react-native';
import React from 'react';
import createStyles from './StatusBar.styles';
import {useTheme} from '~/theming/hooks';

interface Props {
  hidden?: boolean;
}

const StatusBar: React.FC<Props> = ({hidden = false}) => {
  const theme = useTheme();
  const styles = React.useMemo(() => createStyles(theme), [theme]);
  return (
    <RNStatusBar
      backgroundColor={styles.backgroundColor}
      barStyle={styles.barStyle}
      hidden={hidden}
    />
  );
};

export default StatusBar;
