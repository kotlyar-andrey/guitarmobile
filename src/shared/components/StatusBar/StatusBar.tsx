import {StatusBar as RNStatusBar} from 'react-native';
import React from 'react';
import createStyles from './StatusBar.styles';
import {useTheme} from '~/features/themeSwitcher';

interface Props {
  hidden?: boolean;
}

export const StatusBar: React.FC<Props> = ({hidden = false}) => {
  const theme = useTheme();
  const styles = createStyles(theme);
  return (
    <RNStatusBar
      backgroundColor={styles.backgroundColor}
      barStyle={styles.barStyle}
      hidden={hidden}
    />
  );
};
