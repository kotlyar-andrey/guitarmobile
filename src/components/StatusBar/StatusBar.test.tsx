import React from 'react';
import renderer from 'react-test-renderer';

import StatusBar from './StatusBar';

test('StatusBar color test', () => {
  const statusBar = renderer.create(<StatusBar />).toJSON();

  expect(statusBar).toMatchSnapshot();
});
