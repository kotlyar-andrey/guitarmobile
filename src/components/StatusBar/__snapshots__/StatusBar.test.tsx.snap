// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar color test 1`] = `null`;
